### ISSUE TITLE
VERSION_D.html is cached - you must go into incognito

### ISSUE THREAD

**Issue by milanmajchrak** [dataquest-dev/dspace-angular#577]

**Comment by MajoBerger** [comment]
There are several problems interacting with this issue. One or several of those actions can help:
- Refresh with `Ctrl + Shift + R`
- login/logout
- use incognito mode
- switch language to English (or to another language and then back to English)
- wait 10 sec for (anonymous) cache to be dropped and try all of the above again

The problem is, of course, we never know if we see the newest version or not. 
One way would be to mount the file from docker to host and then to statically serve it through e.g. angular.

**Comment by MajoBerger** [comment]
This is a problem for all of static files and possibly for several anonymous pages. I have absolutely no idea what to do about it and seems to be a FE problem.

**Comment by MajoBerger** [comment]
investigate more with languages

**Comment by MajoBerger** [comment]
TL;DR: **To be absolutely sure, use dev tools (F12), check disable cache, and reload.**

VERSION_D.html file (not necessarily the whole page) is cached. 
it is NOT cached in insecure https version - chromium based browsers never cache that.
If `Ctrl + Shift + R` is pressed, the file is requested from the server and not loaded from cache, 
**WITH ONE EXCEPTION**. If language is different, e.g. czech, the server first goes to 
`static/cs/VERSION_D.html` which does not exist, so of course it receives 404, and then goes
to `static/VERSION_D.html` but takes that from cache DESPITE USING `Ctrl + Shift + R`. 
Baffling, but that's what's happening. Using dev tools and checking `disable cache` works 
every time, because no requests are loaded from cache and all are read from server.

When using the original language (in our case english, but I am not sure if that is always
the case or can be configured), `Ctrl + Shift + R` works well and receives the newest version.

Of course, all this can be delayed by 10 seconds (configurable), since frontend server has caching enabled
and user can receive this cached version. This should only be true when not logged in - for anonymous users.
For those that are logged in, there should be no caching on the server side (but above paragraphs still apply! 
This is different caching to the one discussed above!!!!)

**Comment by MajoBerger** [comment]
check how others solve caching of CICD apps

**Comment by MajoBerger** [comment]
when changing versions, files are usually changing names, so cache works as expected. Another solution is to change url by adding parameter that changes, therefore preventing caching.

Adding header is said to be unreliable for cache busting purposes.
